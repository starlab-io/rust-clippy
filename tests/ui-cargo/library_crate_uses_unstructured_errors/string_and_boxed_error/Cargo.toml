[package]
name = "string_and_boxed_error"
version = "0.1.0"
edition = "2021"
publish = false
# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
