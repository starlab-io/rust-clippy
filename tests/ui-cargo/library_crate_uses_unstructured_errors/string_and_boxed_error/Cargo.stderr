error: this is an unstructured error type
 --> src/lib.rs:5:17
  |
5 | pub fn foo() -> Result<(), String> {
  |                 ^^^^^^^^^^^^^^^^^^
  |
  = note: try using an error enum
  = note: `-D clippy::library-crates-structured-errors` implied by `-D warnings`
  = help: to override `-D warnings` add `#[allow(clippy::library_crates_structured_errors)]`

error: this is an unstructured error type
 --> src/lib.rs:9:17
  |
9 | pub fn bar() -> Result<(), Box<dyn Error>> {
  |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: try using an error enum

error: could not compile `string_and_boxed_error` (lib) due to 2 previous errors
